mq/esworker_pubsub.py

```diff
@@ -0,0 +1,185 @@
+            message.ack()
+        except Exception as e:
+            logger.exception(e)
+            logger.error("Malformed message: %r" % message)
+            message.ack()
+
+    def save_event(self, event, metadata):
+        try:
+            # drop the message if a plug in set it to None
+            # signaling a discard
+            if event is None:
+                return
+
+            # make a json version for posting to elastic search
+            jbody = json.JSONEncoder().encode(event)
+
+            try:
+                bulk = False
+                if self.options.esbulksize != 0:
+                    bulk = True
+
+                self.esConnection.save_event(index=metadata["index"], doc_id=metadata["id"], body=jbody, bulk=bulk)
+
+            except (ElasticsearchBadServer, ElasticsearchInvalidIndex) as e:
+                # handle loss of server or race condition with index rotation/creation/aliasing
+                try:
+                    self.esConnection = esConnect()
+                    return
+                except (ElasticsearchBadServer, ElasticsearchInvalidIndex, ElasticsearchException) as e:
+                    logger.exception("ElasticSearchException: {0} reported while indexing event".format(e))
+                    return
+            except ElasticsearchException as e:
+                logger.exception("ElasticSearchException: {0} reported while indexing event".format(e))
+                logger.error("Malformed jbody: %r" % jbody)
+                return
+        except Exception as e:
+            logger.exception(e)
+            logger.error("Malformed message: %r" % event)
+
+
+def esConnect():
+    """open or re-open a connection to elastic search"""
+    return ElasticsearchClient((list("{0}".format(s) for s in options.esservers)), options.esbulksize)
+
+
+def initConfig():
+    # capture the hostname
+    options.mozdefhostname = getConfig("mozdefhostname", socket.gethostname(), options.configfile)
+
+    # elastic search options. set esbulksize to a non-zero value to enable bulk posting, set timeout to post no matter how many events after X seconds.
+    options.esservers = list(getConfig("esservers", "http://localhost:9200", options.configfile).split(","))
+    options.esbulksize = getConfig("esbulksize", 0, options.configfile)
+    options.esbulktimeout = getConfig("esbulktimeout", 30, options.configfile)
+
+    # GCP PubSub options
+    options.resource_name = getConfig("resource_name", "", options.configfile)
+    options.credentials_file = getConfig("credentials_file", "", options.configfile)
+
+    options.mqprotocol = getConfig("mqprotocol", "pubsub", options.configfile)
+
+    # plugin options
+    # secs to pass before checking for new/updated plugins
+    # seems to cause memory leaks..
+    # regular updates are disabled for now,
+    # though we set the frequency anyway.
+    options.plugincheckfrequency = getConfig("plugincheckfrequency", 120, options.configfile)
```

Comment:
This option is no longer used, so we can delete it.

Reply:
Removed